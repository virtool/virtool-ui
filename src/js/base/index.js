export { Accordion } from "./accordion/Accordion";
export { AccordionContent } from "./accordion/AccordionContent";
export { AccordionTrigger } from "./accordion/AccordionTrigger";
export { ScrollingAccordionItem } from "./accordion/ScrollingAccordionItem";
export { Alert } from "./Alert";
export { AlertInner } from "./AlertInner";
export { AlertOuter } from "./AlertOuter";
export { Attribution } from "./Attribution";
export { AttributionWithName } from "./AttributionWithName";
export { Badge } from "./Badge";
export { Box } from "./Box";
export { BoxGroup } from "./BoxGroup";
export { BoxGroupDisabled } from "./BoxGroupDisabled";
export { BoxGroupHeader } from "./BoxGroupHeader";
export { BoxGroupSearch } from "./BoxGroupSearch";
export { BoxGroupSection } from "./BoxGroupSection";
export { BoxGroupTable } from "./BoxGroupTable";
export { BoxSpaced } from "./BoxSpaced";
export { BoxTitle } from "./BoxTitle";
export { device } from "./Breakpoints";
export { Button } from "./Button";
export { Checkbox } from "./Checkbox";
export { CheckboxLabel } from "./CheckboxLabel";
export { Circle } from "./Circle";
export { CloseButton } from "./CloseButton";
export { Color } from "./Color";
export { ComboBox } from "./combobox/ComboBox";
export { Container } from "./Container";
export { ContainerNarrow } from "./ContainerNarrow";
export { ContainerSide } from "./ContainerSide";
export { ContainerWide } from "./ContainerWide";
export { Dialog } from "./Dialog";
export { DialogAlert } from "./DialogAlert";
export { DialogContent } from "./DialogContent";
export { DialogContext } from "./DialogContext";
export { DialogFooter } from "./DialogFooter";
export { DialogOverlay } from "./DialogOverlay";
export { DialogTabs } from "./DialogTabs";
export { DialogTitle } from "./DialogTitle";
export { DialogTrigger } from "./DialogTrigger";
export { DividerHorizontal } from "./DividerHorizontal";
export { DividerVertical } from "./DividerVertical";
export { Dropdown } from "./Dropdown";
export { DropdownButton } from "./DropdownButton";
export { DropdownMenuContent } from "./DropdownMenuContent";
export { DropdownMenuDownload } from "./DropdownMenuDownload";
export { DropdownMenuItem } from "./DropdownMenuItem";
export { DropdownMenuLink } from "./DropdownMenuLink";
export { ExternalLink } from "./ExternalLink";
export { Icon } from "./Icon";
export { IconButton } from "./IconButton";
export { InitialIcon } from "./InitialIcon";
export { Input } from "./Input";
export { InputContainer } from "./InputContainer";
export { InputContext } from "./InputContext";
export { InputError } from "./InputError";
export { InputGroup } from "./InputGroup";
export { InputIcon } from "./InputIcon";
export { InputIconButton } from "./InputIconButton";
export { InputLabel } from "./InputLabel";
export { InputLevel } from "./InputLevel";
export { InputLoading } from "./InputLoading";
export { InputPassword } from "./InputPassword";
export { InputSearch } from "./InputSearch";
export { InputSelect } from "./InputSelect";
export { InputSimple } from "./InputSimple";
export { Key } from "./Key";
export { Label } from "./Label";
export { LinkButton } from "./LinkButton";
export { Loader } from "./Loader";
export { LoadingPlaceholder } from "./LoadingPlaceholder";
export { Logo } from "./Logo";
export { Markdown } from "./Markdown";
export { NoneFound } from "./NoneFound";
export { NoneFoundBox } from "./NoneFoundBox";
export { NoneFoundSection } from "./NoneFoundSection";
export { NotFound } from "./NotFound";
export { Pagination } from "./pagination/Pagination";
export { ProgressBar } from "./ProgressBar";
export { ProgressBarAffixed } from "./ProgressBarAffixed";
export { ProgressCircle } from "./ProgressCircle";
export { RelativeTime } from "./RelativeTime";
export { RemoveBanner } from "./RemoveBanner";
export { SaveButton } from "./SaveButton";
export { SectionHeader } from "./SectionHeader";
export { Select } from "./select/Select";
export { SelectButton } from "./select/SelectButton";
export { SelectContent } from "./select/SelectContent";
export { SelectItem } from "./select/SelectItem";
export { SelectBox } from "./SelectBox";
export { SelectBoxGroupSection } from "./SelectBoxGroupSection";
export { SidebarHeader } from "./SidebarHeader";
export { SidebarHeaderButton } from "./SidebarHeaderButton";
export { SideBarSection } from "./SideBarSection";
export { SlashList } from "./SlashList";
export { SubviewHeader } from "./SubviewHeader";
export { SubviewHeaderAttribution } from "./SubviewHeaderAttribution";
export { SubviewHeaderTitle } from "./SubviewHeaderTitle";
export { Table } from "./Table";
export { Tabs } from "./Tabs";
export { TabsLink } from "./TabsLink";
export { TextArea } from "./TextArea";
export { Toolbar } from "./Toolbar";
export { ViewHeader } from "./ViewHeader";
export { ViewHeaderAttribution } from "./ViewHeaderAttribution";
export { ViewHeaderIcons } from "./ViewHeaderIcons";
export { ViewHeaderSubtitle } from "./ViewHeaderSubtitle";
export { ViewHeaderTitle } from "./ViewHeaderTitle";
