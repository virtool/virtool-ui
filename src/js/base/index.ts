import InputGroup from "@base/InputGroup.js";
import InputLabel from "@base/InputLabel.js";

export { InputGroup, InputLabel };

export { Link } from "wouter";
export { Checkbox } from "./Checkbox";
export { CloseButton } from "./CloseButton";
export { Color } from "./Color";
export { ComboBox } from "./ComboBox";
export { Container } from "./Container";
export { ContainerNarrow } from "./ContainerNarrow";
export { ContainerSide } from "./ContainerSide";
export { ContainerWide } from "./ContainerWide";
export { Dialog } from "./Dialog";
export { DialogAlert } from "./DialogAlert";
export { DialogContent } from "./DialogContent";
export { DialogContext } from "./DialogContext";
export { DialogFooter } from "./DialogFooter";
export { DialogOverlay } from "./DialogOverlay";
export { DialogTabs } from "./DialogTabs";
export { DialogTitle } from "./DialogTitle";
export { DialogTrigger } from "./DialogTrigger";
export { Dropdown } from "./Dropdown";
export { DropdownButton } from "./DropdownButton";
export { DropdownMenuContent } from "./DropdownMenuContent";
export { DropdownMenuDownload } from "./DropdownMenuDownload";
export { DropdownMenuItem } from "./DropdownMenuItem";
export { DropdownMenuLink } from "./DropdownMenuLink";
export { ExternalLink } from "./ExternalLink";
export { Icon } from "./Icon";
export { IconButton } from "./IconButton";
export { InitialIcon } from "./InitialIcon";
export { Input } from "./Input";
export { InputContainer } from "./InputContainer";
export { InputContext } from "./InputContext";
export { InputError } from "./InputError";
export { InputIcon } from "./InputIcon";
export { InputIconButton } from "./InputIconButton";
export { InputLoading } from "./InputLoading";
export { InputPassword } from "./InputPassword";
export { InputSearch } from "./InputSearch";
export { InputSelect } from "./InputSelect";
export { InputSimple } from "./InputSimple";
export { Key } from "./Key";
export { Label } from "./Label";
export { LinkButton } from "./LinkButton";
export { Loader } from "./Loader";
export { LoadingPlaceholder } from "./LoadingPlaceholder";
export { Logo } from "./Logo";
export { Markdown } from "./Markdown";
export { NoneFound } from "./NoneFound";
export { NoneFoundBox } from "./NoneFoundBox";
export { NoneFoundSection } from "./NoneFoundSection";
export { NotFound } from "./NotFound";
export { Pagination } from "./Pagination";
export { ProgressBar } from "./ProgressBar";
export { ProgressBarAffixed } from "./ProgressBarAffixed";
export { ProgressCircle } from "./ProgressCircle";
export { RelativeTime } from "./RelativeTime";
export { RemoveBanner } from "./RemoveBanner";
export { SaveButton } from "./SaveButton";
export { SectionHeader } from "./SectionHeader";
export { Select } from "./Select";
export { SelectBox } from "./SelectBox";
export { SelectBoxGroupSection } from "./SelectBoxGroupSection";
export { SelectButton } from "./SelectButton";
export { SelectContent } from "./SelectContent";
export { SelectItem } from "./SelectItem";
export { SidebarHeader } from "./SidebarHeader";
export { SidebarHeaderButton } from "./SidebarHeaderButton";
export { SideBarSection } from "./SideBarSection";
export { SlashList } from "./SlashList";
export { SubviewHeader } from "./SubviewHeader";
export { SubviewHeaderAttribution } from "./SubviewHeaderAttribution";
export { SubviewHeaderTitle } from "./SubviewHeaderTitle";
export { Table } from "./Table";
export { Tabs } from "./Tabs";
export { TabsLink } from "./TabsLink";
export { TextArea } from "./TextArea";
export { ViewHeader } from "./ViewHeader";
export { ViewHeaderAttribution } from "./ViewHeaderAttribution";
export { ViewHeaderIcons } from "./ViewHeaderIcons";
export { ViewHeaderSubtitle } from "./ViewHeaderSubtitle";
export { ViewHeaderTitle } from "./ViewHeaderTitle";
