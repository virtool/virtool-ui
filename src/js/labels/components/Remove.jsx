import { DialogPortal, DialogTrigger } from "@radix-ui/react-dialog";
import React, { useState } from "react";
import { useMutation, useQueryClient } from "react-query";
import styled from "styled-components";
import { Request } from "../../app/request";
import { getFontSize } from "../../app/theme";
import { Button, Dialog, DialogContent, DialogOverlay, DialogTitle } from "../../base";
import { StyledButtonSmall } from "../../base/styled/StyledButtonSmall";

const RemoveLabelQuestion = styled.p`
    font-size: ${getFontSize("lg")};
`;

const RemoveLabelFooter = styled.footer`
    display: flex;
    margin-top: 30px;
`;

export function RemoveLabel({ id, name }) {
    const [show, setShow] = useState(false);

    const queryClient = useQueryClient();

    const mutation = useMutation(() => {
        return Request.delete(`/api/labels/${id}`);
    });

    const handleDelete = () => {
        mutation.mutate(["label", id], {
            onSuccess: () => {
                setShow(false);
                queryClient.invalidateQueries("labels");
            }
        });
    };

    return (
        <Dialog show={show} onOpenChange={open => setShow(open)}>
            <StyledButtonSmall as={DialogTrigger}>Delete</StyledButtonSmall>
            <DialogPortal>
                <DialogOverlay />
                <DialogContent>
                    <DialogTitle>Delete Label</DialogTitle>
                    <RemoveLabelQuestion>
                        Are you sure you want to delete the label <strong>{name}</strong>?
                    </RemoveLabelQuestion>
                    <RemoveLabelFooter>
                        <Button type="button" color="red" onClick={handleDelete}>
                            Delete
                        </Button>
                    </RemoveLabelFooter>
                </DialogContent>
            </DialogPortal>
        </Dialog>
    );
}
