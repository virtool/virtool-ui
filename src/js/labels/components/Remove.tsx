import { DialogPortal, DialogTrigger } from "@radix-ui/react-dialog";
import { useMutation, useQueryClient } from "@tanstack/react-query";
import React, { useState } from "react";
import styled from "styled-components";
import { Request } from "../../app/request";
import { getFontSize } from "../../app/theme";
import { Button, Dialog, DialogContent, DialogOverlay, DialogTitle } from "../../base";
import { StyledButtonSmall } from "../../base/styled/StyledButtonSmall";
import { labelQueryKeys } from "../queries";

const RemoveLabelQuestion = styled.p`
    font-size: ${getFontSize("lg")};
`;

const RemoveLabelFooter = styled.footer`
    display: flex;
    margin-top: 30px;
`;

interface RemoveLabelProps {
    id: string;
    name: string;
}

export function RemoveLabel({ id, name }: RemoveLabelProps) {
    const [open, setOpen] = useState(false);

    const queryClient = useQueryClient();

    const mutation = useMutation(
        () => {
            return Request.delete(`/labels/${id}`);
        },
        {
            onSuccess: () => {
                setOpen(false);
                queryClient.invalidateQueries(labelQueryKeys.lists());
            },
        },
    );

    return (
        <Dialog open={open} onOpenChange={open => setOpen(open)}>
            <StyledButtonSmall as={DialogTrigger}>Delete</StyledButtonSmall>
            <DialogPortal>
                <DialogOverlay />
                <DialogContent>
                    <DialogTitle>Delete Label</DialogTitle>
                    <RemoveLabelQuestion>
                        Are you sure you want to delete the label <strong>{name}</strong>?
                    </RemoveLabelQuestion>
                    <RemoveLabelFooter>
                        <Button type="button" color="red" onClick={mutation.mutate}>
                            Delete
                        </Button>
                    </RemoveLabelFooter>
                </DialogContent>
            </DialogPortal>
        </Dialog>
    );
}
