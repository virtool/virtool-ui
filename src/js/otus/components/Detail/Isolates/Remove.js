import { get } from "lodash-es";
import React, { useCallback } from "react";
import { connect } from "react-redux";
import { RemoveModal } from "../../../../base";
import { hideOTUModal, removeIsolate } from "../../../actions";

export const Remove = ({ id, name, nextId, otuId, show, onConfirm, onHide }) => {
    const handleConfirm = useCallback(() => {
        onConfirm(otuId, id, nextId);
    }, [otuId, id]);

    return <RemoveModal name={name} noun="Isolate" onConfirm={handleConfirm} onHide={onHide} show={show} />;
};

export const mapStateToProps = state => {
    const { id, name } = state.otus.activeIsolate;
    return {
        id,
        name,
        nextId: get(state, ["otus", "detail", "isolates", 0, "id"], null),
        otuId: state.otus.detail.id,
        show: state.otus.removeIsolate
    };
};

export const mapDispatchToProps = dispatch => ({
    onHide: () => {
        dispatch(hideOTUModal());
    },

    onConfirm: (otuId, isolateId, nextIsolateId) => {
        dispatch(removeIsolate(otuId, isolateId, nextIsolateId));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(Remove);
