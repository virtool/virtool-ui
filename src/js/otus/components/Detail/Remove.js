import React, { useCallback } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { removeOTU, hideOTUModal } from "../../actions";
import { RemoveModal } from "../../../base";

export const Remove = ({ history, id, name, refId, show, onConfirm, onHide }) => {
    const handleConfirm = useCallback(() => {
        onConfirm(refId, id, history);
    }, [id, refId]);

    return <RemoveModal name={name} noun="OTU" onConfirm={handleConfirm} onHide={onHide} show={show} />;
};

Remove.propTypes = {
    history: PropTypes.object,
    id: PropTypes.string,
    name: PropTypes.string,
    refId: PropTypes.string,
    show: PropTypes.bool,
    onConfirm: PropTypes.func,
    onHide: PropTypes.func
};

export const mapStateToProps = state => ({
    show: state.otus.remove,
    refId: state.references.detail.id
});

export const mapDispatchToProps = dispatch => ({
    onConfirm: (refId, otuId, history) => {
        dispatch(removeOTU(refId, otuId, history));
    },

    onHide: () => {
        dispatch(hideOTUModal());
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(Remove);
