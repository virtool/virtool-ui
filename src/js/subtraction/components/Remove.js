import React from "react";
import { connect } from "react-redux";
import { pushState } from "../../app/actions";
import { RemoveModal } from "../../base";
import { routerLocationHasState } from "../../utils/utils";
import { removeSubtraction } from "../actions";

export const RemoveSubtraction = ({ id, name, show, onConfirm, onHide }) => (
    <RemoveModal id={id} name={name} noun="Subtraction" show={show} onHide={onHide} onConfirm={() => onConfirm(id)} />
);

export const mapStateToProps = state => ({
    id: state.subtraction.detail.id,
    name: state.subtraction.detail.name,
    show: routerLocationHasState(state, "removeSubtraction", true)
});

export const mapDispatchToProps = dispatch => ({
    onHide: () => {
        dispatch(pushState({ removeSubtraction: false }));
    },

    onConfirm: subtractionId => {
        dispatch(removeSubtraction(subtractionId));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(RemoveSubtraction);
