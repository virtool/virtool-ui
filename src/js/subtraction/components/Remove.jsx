import React from "react";
import { connect } from "react-redux";
import { pushState } from "../../app/actions";
import { RemoveModal } from "../../base";
import { routerLocationHasState } from "../../utils/utils";
import { removeSubtraction } from "../actions";

export const RemoveSubtraction = ({ subtraction, show, onConfirm, onHide }) => (
    <RemoveModal
        id={subtraction.id}
        name={subtraction.name}
        noun="Subtraction"
        show={show}
        onHide={onHide}
        onConfirm={() => onConfirm(subtraction.id)}
    />
);

export const mapStateToProps = state => ({
    show: routerLocationHasState(state, "removeSubtraction", true),
});

export const mapDispatchToProps = dispatch => ({
    onHide: () => {
        dispatch(pushState({ removeSubtraction: false }));
    },

    onConfirm: subtractionId => {
        dispatch(removeSubtraction(subtractionId));
    },
});

export default connect(mapStateToProps, mapDispatchToProps)(RemoveSubtraction);
