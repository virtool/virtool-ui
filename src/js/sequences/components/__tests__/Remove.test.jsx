import { shallow } from "enzyme";
import React from "react";
import { beforeEach, describe, expect, it, vi } from "vitest";
import { RemoveModal } from "../../../base";
import { RemoveSequence } from "../Remove";

describe("<RemoveSequence />", () => {
    let props;

    beforeEach(() => {
        props = {
            otuId: "foo",
            isolateId: "bar",
            isolateName: "baz",
            sequenceId: "test",
            onHide: vi.fn(),
            onConfirm: vi.fn(),
        };
    });

    it("renders correctly", () => {
        const wrapper = shallow(<RemoveSequence {...props} />).dive();
        expect(wrapper).toMatchSnapshot();
    });

    it("calls onConfirm when remove confirmed", () => {
        const wrapper = shallow(<RemoveSequence {...props} />);
        wrapper.find(RemoveModal).prop("onConfirm")();
        expect(props.onConfirm).toHaveBeenCalledWith("foo", "bar", "test");
    });

    it("calls onHide when hidden", () => {
        const wrapper = shallow(<RemoveSequence {...props} />);
        wrapper.find(RemoveModal).prop("onHide")();
        expect(props.onHide).toHaveBeenCalled();
    });
});
