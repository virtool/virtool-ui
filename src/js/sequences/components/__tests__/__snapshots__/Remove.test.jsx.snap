// Vitest Snapshot v1

exports[`<RemoveSequence /> > renders correctly 1`] = `
<Modal
  color="red"
  label="Remove"
  onHide={[MockFunction spy]}
  show={true}
  size="sm"
>
  <ModalHeader>
    Remove Sequence
  </ModalHeader>
  <RemoveModalBody>
    <span>
      Are you sure you want to remove the sequence
      <strong>
         
        test
      </strong>
       from
      <strong>
         
        baz
      </strong>
      ?
    </span>
  </RemoveModalBody>
  <ModalFooter>
    <Button
      color="red"
      icon="check"
      onClick={[Function]}
    >
      Confirm
    </Button>
  </ModalFooter>
</Modal>
`;
